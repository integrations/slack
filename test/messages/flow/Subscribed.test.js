const { Subscribed } = require('../../../lib/messages/flow');
const repoFixture = require('../../fixtures/repo.json');

describe('Subscribed message rendering', () => {
  test('works for successfully subscribed', async () => {
    const message = new Subscribed({
      resource: repoFixture,
    });
    expect(message.toJSON()).toMatchSnapshot();
  });

  test('works for successfully unsubscribed', async () => {
    const message = new Subscribed({
      resource: repoFixture,
      unsubscribed: true,
    });
    expect(message.toJSON()).toMatchSnapshot();
  });
});
