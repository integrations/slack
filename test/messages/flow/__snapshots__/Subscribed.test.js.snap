// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subscribed message rendering works for successfully subscribed 1`] = `
Object {
  "attachments": Array [
    Object {
      "color": "#24292f",
      "text": "Subscribed to <https://github.com/bkeepers/dotenv|bkeepers/dotenv>",
    },
  ],
  "response_type": "in_channel",
}
`;

exports[`Subscribed message rendering works for successfully unsubscribed 1`] = `
Object {
  "attachments": Array [
    Object {
      "color": "#24292f",
      "text": "Unsubscribed from <https://github.com/bkeepers/dotenv|bkeepers/dotenv>",
    },
  ],
  "response_type": "in_channel",
}
`;
